android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt

```diff
@@ -986,6 +986,23 @@ abstract class EngineSession(
      */
     abstract fun requestTranslationRestore()
 
+    /**
+     * Requests the [EngineSession] retrieve the current site's translation preference.
+     */
+    abstract fun getNeverTranslateSiteSetting(
+        onResult: (Boolean) -> Unit,
+        onException: (Throwable) -> Unit,
+    )
+
+    /**
+     * Requests the [EngineSession] to set the current site's translation preference.
+     */
+    abstract fun setNeverTranslateSiteSetting(
+        setting: Boolean,
+        onResult: () -> Unit,
+        onException: (Throwable) -> Unit,
+    )
```

Comment:
Similar to https://github.com/mozilla-mobile/firefox-android/pull/4628/files#r1411126935

Reply:
Updated, thanks!